internal/store/data_federation.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	atlas "go.mongodb.org/atlas/mongodbatlas"
```

Comment:
any reason not to use the new SDK? we should stop using this package

Reply:
Old SDK was used as it seemed to be standardly used in kubernetes side of the CLI and is used in Atlas K8s Operator.

Looking at the new SDK, it's api_data_federation uses DataLakeTenant but this seems to contain all same fields as the Data Federation stuct used by AKO, so fix should be relatively painless.

[This file](https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/store/atlas/data_federation.go) can be used as an alternative to store/data_federation. Fix will be needed to address that interface `AtlasOperatorGenericStore` cannot access methods in another package. store/atlas/data_federation will either have to move into store, or store/operator into store/atlas. There may be an alternative fix I'm not seeing.

Reply:
investigating the SDK change

Reply:
Not sure if the evergreen issues are related to this changes or not.

Reply:
I saw a green in evergreen. Closing for now.